src/conn/conn_compact.c

```diff
@@ -21,6 +21,92 @@ __background_compact_server_run_chk(WT_SESSION_IMPL *session)
     return (FLD_ISSET(S2C(session)->server_flags, WT_CONN_SERVER_COMPACT));
 }
 
+/*
+ * __background_compact_exclude_list_add --
+ *     Add the entry to the exclude hash table.
+ */
+static int
+__background_compact_exclude_list_add(WT_SESSION_IMPL *session, const char *name, size_t len)
+{
+    WT_BACKGROUND_COMPACT_EXCLUDE *new_entry;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    uint64_t bucket, hash;
+
+    conn = S2C(session);
+
+    WT_ERR(__wt_calloc_one(session, &new_entry));
+    WT_ERR(__wt_strndup(session, name, len, &new_entry->name));
+
+    hash = __wt_hash_city64(name, len);
+    bucket = hash & (conn->hash_size - 1);
+    /* Insert target uri entry into hashtable. */
+    TAILQ_INSERT_HEAD(&conn->background_compact.exclude_list_hash[bucket], new_entry, hashq);
+
+    return (0);
+err:
+    if (new_entry != NULL)
+        __wt_free(session, new_entry->name);
+    __wt_free(session, new_entry);
+
+    return (ret);
+}
+
+/*
+ * __background_compact_exclude_list_clear --
+ *     Clear the list of entries excluded from compaction.
+ */
+static void
+__background_compact_exclude_list_clear(WT_SESSION_IMPL *session, bool closing)
+{
+    WT_BACKGROUND_COMPACT_EXCLUDE *entry;
+    WT_CONNECTION_IMPL *conn;
+    uint64_t i;
+
+    conn = S2C(session);
+
+    for (i = 0; i < conn->hash_size; ++i) {
+        while (!TAILQ_EMPTY(&conn->background_compact.exclude_list_hash[i])) {
+            entry = TAILQ_FIRST(&conn->background_compact.exclude_list_hash[i]);
+            /* Remove target uri entry from the hashtable. */
+            TAILQ_REMOVE(&conn->background_compact.exclude_list_hash[i], entry, hashq);
+            __wt_free(session, entry->name);
+            __wt_free(session, entry);
+        }
+    }
+
+    if (closing)
```

Comment:
Instead here I could check against `F_ISSET(S2C(session), WT_CONN_CLOSING)` but I like having the arg too.